specs/flyout/flyout.md

```diff
@@ -0,0 +1,168 @@
+# Flyout
+
+## Overview
+Flyout is a focusable floating container that shows over a pages content and displays UI related to what the user is doing. Similar to a Dialog, but with positioning controls and no specific action is required, a Flyout can be used to reveal a secondary control or show more detail about an item.
+
+### Background
+A Flyout can be less intrusive to the user experience than a Dialog and should be used instead of a Dialog when information or input is needed to be displayed, but not in the center of the screen, such as in the case of collection of additional info before an action, displaying information that is only relevant some of the time, or displaying more information.
+
+### Use Cases
+- As a Dropdown for a menu, navigation bar, etc.
+- In a context menu as the secondary menu/action.
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyouts anchor element. This way it can be dynamic and adapt to the available space if `responsive` is true for the Flyout.
+
+**Light Dismiss (aka Soft Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key.
+
+**Overlay:** - A control of whether or not the Flyout has an overlay effect that dims the surrounding UI when the Flyout is active.
```

Comment:
How common is this with other instances of flyout-like controls across libraries? This seems more common with a dialog pattern, whereas things like flyout, callout, coachmarks seem to rarely have overlays. If this is common, can you link to some examples here?

Reply:
Just browsing around for new features, but I have an example I find quite common - it'd be for tutorial/intro/demo interactions where a user is guided to perform actions, sometimes not allowed to proceed without completing required action.